internal/repository/event.go

```diff
@@ -48,6 +51,9 @@ type ListEventOpts struct {
 
 	// (optional) the order direction
 	OrderDirection *string `validate:"omitempty,oneof=ASC DESC"`
+
+	// (optional) the event metadata
+	AdditionalMetadata *[]byte
```

Comment:
```suggestion
	AdditionalMetadata []byte
```

Arrays are nil by default, so `*` is unnecessary